Provide better logging/exception when invalid credentials are passed in

If bad credentials are set, then the stack trace is several hundred lines long  (see attached stacktrace.txt) with the only meaningful bits being

```
Caused by: org.cloudfoundry.uaa.UaaException: unauthorized: Bad credentials
```

and 

```
[org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryDeployerAutoConfiguration$EarlyConnectionConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.github.zafarkhaja.semver.Version]: Factory method 'version' threw exception; nested exception is org.cloudfoundry.uaa.UaaException: unauthorized: Bad credentials
```

and

```
org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryDeployerAutoConfiguration$EarlyConnectionConfiguration.version(CloudFoundryDeployerAutoConfiguration.java:170) ~[spring-cloud-deployer-cloudfoundry-1.4.4.RELEASE.jar!/:1.4.4.RELEASE]
```

but is hard to see.  

The code to improve is this

```
		@Bean
		@ConditionalOnMissingBean
		public Version version(CloudFoundryClient client) {
			return client.info()
				.get(GetInfoRequest.builder()
					.build())
				.map(response -> Version.valueOf(response.getApiVersion()))
				.doOnNext(version -> logger.info("Connecting to Cloud Foundry with API Version {}", version))
				.block(Duration.ofSeconds(appDeploymentProperties().getApiTimeout()));
		}
```
If there an exception thrown, it looks like it doesn't get triggered until `Version.valueOf(response.getApiVersion()` 
is executed.  We should check the response and not try to do any processing if it isn't successful.  Also, we should catch the exception and print out a short sweet message about credentials being invalid.

Should backport as well.

--- a/src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryDeployerAutoConfiguration.java
+++ b/src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryDeployerAutoConfiguration.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016 the original author or authors.
+ * Copyright 2016-2019 the original author or authors.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -30,6 +30,7 @@ import org.cloudfoundry.reactor.DefaultConnectionContext;
 import org.cloudfoundry.reactor.TokenProvider;
 import org.cloudfoundry.reactor.client.ReactorCloudFoundryClient;
 import org.cloudfoundry.reactor.tokenprovider.PasswordGrantTokenProvider;
+import org.cloudfoundry.uaa.UaaException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -170,6 +171,9 @@ public class CloudFoundryDeployerAutoConfiguration {
 				.get(GetInfoRequest.builder()
 					.build())
 				.map(response -> Version.valueOf(response.getApiVersion()))
+				.doOnError(e -> {
+					throw new RuntimeException("Bad credentials connecting to Cloud Foundry.", e);
+				})
 				.doOnNext(version -> logger.info("Connecting to Cloud Foundry with API Version {}", version))
 				.block(Duration.ofSeconds(appDeploymentProperties().getApiTimeout()));
 		}

